suricata/update/net.py

```diff
@@ -111,25 +122,37 @@ def get(url, fileobj, progress_hook=None):
     else:
         opener.addheaders = [(header, value) for header,
                              value in opener.addheaders if header.lower() != "user-agent"]
-    remote = opener.open(url)
-    info = remote.info()
+    if isinstance(url, tuple):
+        header = url[1].split(":") if url[1] is not None else None
+        if header and is_header_clean(header=header):
+            logger.debug("Setting HTTP header %s to %s", header[0], header[1])
+            opener.addheaders = [(header[0], header[1]), ]
+        elif header:
+            logger.error("Header not set as it does not meet the criteria")
+        url = url[0]
+
```

Comment:
When using --header "X-Key: XXX", I noticed that the server actually receives 2 spaces between the `:` and the `XXX`. header[1] probably needs to be stripped here.